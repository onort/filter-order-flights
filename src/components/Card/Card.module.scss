@import "../../styles/main.scss";

.container {
  display: block;
  width: 100%;
  background-color: $white;
  border-radius: px-to-rem(2px);
  box-shadow: 0 1px 5px rgba(0, 0, 0, 0.233752);
}
