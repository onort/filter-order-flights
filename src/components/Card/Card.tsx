import React from "react"
import cx from "classnames"

import styles from "./Card.module.scss"

interface Props {
  className?: string
}

const Card: React.SFC<Props> = props => {
  const classNames = cx(styles.container, props.className)
  return <div className={classNames}>{props.children}</div>
}

export default Card
