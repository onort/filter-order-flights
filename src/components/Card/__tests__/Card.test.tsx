import React from "react"
import { shallow } from "enzyme"

import Card from "../"

describe("<Card />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Card>
        <span>Test content.</span>
      </Card>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have a custom classname", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <Card className={testClassName}>
        <span>Test content.</span>
      </Card>
    )
    expect(wrapper.find("div").hasClass(testClassName)).toBe(true)
  })
})
