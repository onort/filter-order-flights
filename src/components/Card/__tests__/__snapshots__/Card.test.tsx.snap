// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    <span>
      Test content.
    </span>
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        Test content.
      </span>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test content.",
      },
      "ref": null,
      "rendered": "Test content.",
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          Test content.
        </span>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test content.",
        },
        "ref": null,
        "rendered": "Test content.",
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Card /> should match snapshot 2`] = `"<div class=\\"container\\"><span>Test content.</span></div>"`;
